crossplane/terrajet
# Pipeline abstraction and configuration interface improvements

### Description of your changes

This PR abstracts the terrajet code running generation pipelines from tf providers. With the changes here, terrajet based providers would only need to provide a `Provider` configuration (with resource-specific configurations) and Terrajet will take care of the rest.

This PR also makes some (breaking) changes to interact with Terrajet aiming for a more flexible (and hopefully more stable) interface. 

Finally adds support for generating Terraform Providers depending on an old version (<v2) of Terraform plugin SDK.

Fixes #61
Fixes #129 

I have:


### How has this code been tested

Tested with `autoscaling` and `ec2` groups in provider-tf-aws using [this PR](https://github.com/crossplane-contrib/provider-tf-aws/pull/126/):

1. Run `make generate`
2. Verify that there is no unexpected changes in the generated code. 

[contribution process]: https://git.io/fj2m9
